xbmc/epg/EpgContainer.cpp

```diff
@@ -235,20 +235,16 @@ void CEpgContainer::LoadFromDB(void)
 
 bool CEpgContainer::PersistTables(void)
 {
-  m_critSection.lock();
-  std::map<unsigned int, CEpg*> copy = m_epgs;
-  m_critSection.unlock();
-  return m_database.Persist(copy);
+  CSingleLock lock(m_critSection);
+  return m_database.Persist(m_epgs);
 }
 
 bool CEpgContainer::PersistAll(void)
 {
   bool bReturn(true);
-  m_critSection.lock();
-  std::map<unsigned int, CEpg*> copy = m_epgs;
-  m_critSection.unlock();
+  CSingleLock lock(m_critSection);
 
-  for (EPGMAP_CITR it = copy.begin(); it != copy.end() && !m_bStop; it++)
+  for (EPGMAP_CITR it = m_epgs.begin(); it != m_epgs.end() && !m_bStop; it++)
```

Comment:
it would be nice if you move these changes in a separate commit or PR because we need to wait for a response of @opdenkamp who knows why we leave the lock before calling the database stuff.